src/state-item.c

```diff
@@ -0,0 +1,417 @@
+//
+//  state-item.c
+//  bison
+//
+//  Created by Vincent Imbimbo on 10/15/19.
+//  Copyright © 2019 Vincent Imbimbo. All rights reserved.
+//
+
+#include <config.h>
+#include <stdlib.h>
+#include "state-item.h"
+#include "closure.h"
+#include "nullable.h"
+#include <time.h>
+#include <gl_linked_list.h>
+#include <gl_xlist.h>
+#include <assert.h>
+#include "getargs.h"
+
+size_t nstate_items;
+state_item_number *state_item_map;
+state_item *state_items;
+
+state_item_number *trans;
+bitsetv rev_trans;
+Hash_table *prods;
+Hash_table *rev_prods;
+
+
+// hash functions for index -> bitset hash maps
+typedef struct
+{
+    int key;
```

Comment:
You're indenting by 4, Bison is indented by 2.  Please, follow the same patterns as the rest of code.

I'm sorry we don't have a `.clang-format` file, we definitely should.